### PR TITLE
Refactor/dry up hide show

### PR DESCRIPTION
**Pull request by knpausch**
Summary of the work done:

- This incorporates Heather's feedback on DRYing up all hide/show functions so it takes in an array or element variables, iterates through them to add/remove classes

This branch includes:

- refactors hide and show function to take in a parm of an array of elements, iterates through them to add/remove classes
- for each page view that invokes hide/show, local const variables were created to store element variables into an array so it may be passed through hide and show functions

Contributed to code:

- Kiko

Needs to review:

- Lauren
- Eleanor

Additional notes about any issues/questions remaining around this code:

- Run my branch to make sure everything looks smooth

Indicate anything that still needs to be addressed

- I'm 80% confident this may cause a conflict merge since it touches almost all functions with hide/show function invokes

### CODE REVIEW

**Review by FrazierLE — APPROVED** [review]
Much cleaner than what we had before!

File: src/scripts.js [hunk 1]
```diff
@@ -95,27 +95,19 @@ function randomizeUser(data) {
...
-    hide(singleRecipe)
-    show(favoritesView)
-    hide(favoriteRecipeButton)
-    hide(favoriteButton)
-    show(filterSidebar)
-    hide(ingredientSidebar)
+    const hideElementsList = [removeRecipeButton, allRecipes, singleRecipe, favoriteRecipeButton, favoriteButton, ingredientSidebar]
+    const showElementsList = [favoritesView, filterSidebar]
+    hide(hideElementsList)
+    show(showElementsList)
```
Comment (Eleanorgruth) [inline src/scripts.js:110]: I think we can dry this up and don't need the variables and just have `hide([removeRecipeButton, allRecipes, singleRecipe, favoriteRecipeButton, favoriteButton, ingredientSidebar])`
and `show([favoritesView, filterSidebar])`

Comment (Eleanorgruth) [reply src/scripts.js:110]: Same goes for all the others, it seems like this variable is function scoped and we aren't using it elsewhere

**Review by Eleanorgruth — CHANGES_REQUESTED** [review]
I like this change a lot, it cleans up a lot of code. I think we should just dry it up a little more

**Pull request merged**